crates/kornia-core/src/storage.rs

```diff
@@ -160,6 +160,54 @@ where
         }
     }
 
+    /// Creates a new `TensorStorage` from a slice of data.
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - A slice containing the data to be stored.
+    /// * `alloc` - The allocator to use for creating the storage.
+    ///
+    /// # Returns
+    ///
+    /// A new `TensorStorage` instance containing a copy of the input data.
+    ///
+    /// # Errors
+    ///
+    /// Returns a `TensorAllocatorError` if the allocation fails.
+    pub fn from_slice(data: &[T], alloc: A) -> Self {
+        let buffer = Buffer::from_slice_ref(data);
```

Comment:
@emilmgeorge discovered this function, i thinks it still makes copy but we can explore later more safer ways to achieve zero copies. Another alternative could be to explore `from_bytes` https://docs.rs/arrow-buffer/latest/arrow_buffer/buffer/struct.Buffer.html#method.from_bytes

Reply:
👍🏼  (also: need to check how to handle custom allocations for non-CPU allocators?).